packages/@aws-cdk/aws-ec2-alpha/lib/subnet-v2.ts

```diff
@@ -276,6 +283,135 @@ export class SubnetV2 extends Resource implements ISubnetV2 {
+   */
+  readonly ipv4CidrBlock: string;
+
+  /**
+   * The IPv4 CIDR block associated with the subnet
+   *
+   * @default - No CIDR information, cannot use CIDR filter features
+   */
+  readonly ipv6CidrBlock?: string;
+
+  /**
+   * The ID of the route table for this particular subnet
+   *
+   * @default - No route table information, cannot create VPC endpoints
+   */
+  readonly routeTableId?: string;
+
+  /**
+   * The subnetId for this particular subnet
+   */
+  readonly subnetId: string;
+
+  /**
+   * The type of subnet (public or private) that this subnet represents.
+   */
+  readonly subnetType: SubnetType;
+
+  /**
+   * Name of the given subnet
+   *
+   * @default - no subnet name
+   */
+  readonly subnetName?: string;
+
+}
+
+/**
+ * Properties required to import a subnet
+ */
+export interface ImportedSubnetV2Props extends SubnetV2Attributes {}
+
+/**
+ * Class to define an import for an existing subnet
+ * @resource AWS::EC2::Subnet
+ */
+export class ImportedSubnetV2 extends Resource implements ISubnetV2 {
+
+  /**
+   * The IPv6 CIDR Block assigned to this subnet
+   */
+  public readonly ipv6CidrBlock?: string;
+
+  /**
+   * The type of subnet (eg. public or private) that this subnet represents.
+   */
+  public readonly subnetType?: SubnetType;
+
+  /**
+   * The Availability Zone in which subnet is located
+   */
+  public readonly availabilityZone: string;
+
+  /**
+  * The subnetId for this particular subnet
+  * Refers to the physical ID created
+  */
+  public readonly subnetId: string;
+
+  /**
+   * Dependable that can be depended upon to force internet connectivity established on the VPC
+   */
+  public readonly internetConnectivityEstablished: IDependable = new DependencyGroup();
+
+  /**
+   * The IPv4 CIDR block assigned to this subnet
+   */
+  public readonly ipv4CidrBlock: string;
+
+  /**
+  *  Current route table associated with this subnet
+  */
+  public readonly routeTable: IRouteTable;
+
+  constructor(scope: Construct, id: string, props: ImportedSubnetV2Props) {
+    super(scope, id);
+
+    if (!props.routeTableId) {
+      throw new Error('Route Table ID is required');
+    }
+
+    this.ipv4CidrBlock = props.ipv4CidrBlock;
+    this.availabilityZone = props.availabilityZone;
+    this.subnetType = props.subnetType;
+    this.ipv6CidrBlock = props.ipv6CidrBlock;
+    this.subnetId = props.subnetId;
+    this.routeTable = {
+      routeTableId: props.routeTableId!,
+    };
+  }
+
+  /**
+   * Associate a Network ACL with this subnet
+   * Required here since it is implemented in the ISubnetV2
+   */
+  public associateNetworkAcl(id: string, networkAcl: INetworkAcl) {
+    const scope = networkAcl instanceof Construct ? networkAcl : this;
+    const other = networkAcl instanceof Construct ? this : networkAcl;
+    new SubnetNetworkAclAssociation(scope, id + Names.nodeUniqueId(other.node), {
+      networkAcl,
+      subnet: this,
+    });
+  }
```

Comment:
usually the imported class should extends the resource class, and depends on the functions implementation exist in the basic resource class. Why is this class different?